### DISCUSSION TITLE
(Phalcon >= 5.0.0) Router: action name to be camelized like controller name

### DISCUSSION THREAD

**Discussion by borisdelev** [phalcon/cphalcon#15999] [Ideas]
Hi there :)

I want to discuss something about Phalcon router. Its related with action names. In documentation is mention that:
> Controller names are camelized, this means that characters (-) and (_) are removed and the next character is uppercased. For instance, some_controller is converted to SomeController.

Thats great. But what about action names? Lets consider i have url like </some-controller/some-action> ? The router search action name "some-action" in controller "SomeController"... and that is not possible action name, because its function. To work like i want every time i describe in router (example):

```php
$router->add('/:controller/:action/([a-zA-Z0-9\_\-]+)', [
    'controller' => 1,
    'action' => 2,
    'slug' => 3
])->convert('action', function($action) {
    return lcfirst($this->get('helper')->camelize($action));
});
```
... and now my router know- SomeController / someAction. I think is good to have (rule) default action name like ControllerName (of course can be changed via custom rule). Not to mention what view file must be loaded.

Anyway- my idea is simple- default framework rule about actionNames in controllers. What you are thinking?

PS: sorry for my English

**Comment by s-ohnishi** [comment]
Hi.

I also think that forcing the controller to be replaced with "upper"camelCase (=PascalCase) is not balanced.
However, I'm a bit skeptical that actions should also be replaced with "lower"camelCase.

I find it tempting to use complex names for actions rather than controllers, so it's acceptable to want to use symbols to separate things clearly.
However, it is not the default process, but the application side that "want to use symbols", so I think the application should deal with it on its own.
There is a way to do that (just like you do).

However, I think it's a problem that symbols are accepted by default as action strings, but not directly as method names.
